scripts/router.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+mongoAdminUser=$1
+mongoAdminPasswd=$2
+fqdn=$3
+
+install_mongo3() {
+
+	#install
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-mongos-3.6.17-1.el7.x86_64.rpm
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	wget https://repo.mongodb.org/yum/redhat/7/mongodb-org/3.6/x86_64/RPMS/mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-mongos-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-shell-3.6.17-1.el7.x86_64.rpm
+	rpm -i mongodb-org-server-3.6.17-1.el7.x86_64.rpm
+	PATH=$PATH:/usr/bin; export PATH
+
+	#disable selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/sysconfig/selinux
+	sed -i 's/^SELINUX=.*/SELINUX=disabled/' /etc/selinux/config
+	setenforce 0
+
+	#kernel settings
+	if [[ -f /sys/kernel/mm/transparent_hugepage/enabled ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/enabled
+	fi
+	if [[ -f /sys/kernel/mm/transparent_hugepage/defrag ]];then
+		echo never > /sys/kernel/mm/transparent_hugepage/defrag
+	fi
+
+	#configure
+	sed -i 's/\(bindIp\)/#\1/' /etc/mongod.conf
+
+	#set keyfile
+	echo "vfr4CDE1" > /etc/mongokeyfile
+	chown mongod:mongod /etc/mongokeyfile
+	chmod 600 /etc/mongokeyfile
+	sed -i 's/^#security/security/' /etc/mongod.conf
+	sed -i '/^security/akeyFile: /etc/mongokeyfile' /etc/mongod.conf
+	sed -i 's/^keyFile/  keyFile/' /etc/mongod.conf
+}
+
+yum install wget -y
+echo "Generating ssl certificate"
+openssl req -newkey rsa:2048 -nodes -keyout /etc/key.pem -x509 -days 365 -out /etc/certificate.pem -subj "/CN=$fqdn"
+openssl pkcs12 -inkey /etc/key.pem -in /etc/certificate.pem -export -out /etc/MongoAuthCert.p12 -passout pass:"Mongo123"
+openssl pkcs12 -in /etc/MongoAuthCert.p12 -out /etc/MongoAuthCert.pem -passin pass:"Mongo123" -passout pass:"Mongo123"
+
+install_mongo3
+
+#start router server
+mongos --configdb crepset/10.0.0.240:27019,10.0.0.241:27019,10.0.0.242:27019 --bind_ip 0.0.0.0 --port 27017 --logpath /var/log/mongodb/mongos.log --fork --keyFile /etc/mongokeyfile --sslMode requireSSL --sslPEMKeyFile /etc/MongoAuthCert.pem --sslPEMKeyPassword Mongo123
+
+
+#check router server starts or not
+for((i=1;i<=3;i++))
+do
+	sleep 30
+	n=`ps -ef |grep "mongos --configdb" |grep -v grep |wc -l`
+	if [[ $n -eq 1 ]];then
+		echo "mongos started successfully"
+		break
+	else
+		mongos --configdb crepset/10.0.0.240:27019,10.0.0.241:27019,10.0.0.242:27019 --bind_ip 0.0.0.0 --port 27017 --logpath /var/log/mongodb/mongos.log --fork --keyFile /etc/mongokeyfile --sslMode requireSSL --sslPEMKeyFile /etc/MongoAuthCert.pem --sslPEMKeyPassword Mongo123
+		continue
+	fi
+done
+
+n=`ps -ef |grep "mongos --configdb" |grep -v grep |wc -l`
+if [[ $n -ne 1 ]];then
+echo "mongos tried to start 3 times but failed!"
+fi
```

Comment:
exit the script if mongos does not start?

Reply:
I actually noticed the original author wrote the script in a way that it'd always succeed no matter what. I am going to exit with status failed if mongo does not start in all scripts. This way the deployment will fail if mongo instance does not start.

Reply:
Done